### DISCUSSION TITLE
apk file is not output（apkファイルが出力されない）

### DISCUSSION THREAD

**Discussion by HirokatsuIshikawa** [airsdk/Adobe-Runtime-Support#2824] [Q&A] [answered]
Since I changed my PC to a new one, I also changed the environment used for FlashDevelop to a new one, but the PC version of AIR worked fine, but the Android version did not output properly.
Some of the problems have been resolved by looking at the discussion so far, but now the following message appears after executing PackageApp.bat.

//////
Packaging: dist\NeoSRCMobile-captive-runtime.apk
using certificate: cert\NeoSRCMobile.p12...

List of devices attached
{DeviceName} device

Installing dist\NeoSRCMobile-captive-runtime.apk on the device...

Performing Streamed Install
adb: failed to stat dist\NeoSRCMobile-captive-runtime.apk: No such file or directory

Installing the app on the device failed
//////

There are no errors or warnings, but the APK is not output to dist.
The version of AIRSDK is 50.2.3, and for FLEX_SDK, which I could not find, I downloaded the latest version of AIRSDK_Flex_Windows and configured it as follows, but was this okay?

set FLEX_SDK=E:\AIRSDK\AIRSDK_Flex_Windows

※日本語原文
PCを新しいものに変えた為、FlashDevelopに使用する環境も一通り新しいものに変えたのですが、PC版のAIRの方はうまくいったのですがAndroid版の方の出力がうまくいきません。
ここまでのディスカッションを見ていくつかの不具合は解決したのですが、現在はPackageApp.bat実行後に、以下のメッセージが出るようになっております。

//////
Packaging: dist\NeoSRCMobile-captive-runtime.apk
using certificate: cert\NeoSRCMobile.p12...

List of devices attached
{DeviceName}      device

Installing dist\NeoSRCMobile-captive-runtime.apk on the device...

Performing Streamed Install
adb: failed to stat dist\NeoSRCMobile-captive-runtime.apk: No such file or directory

Installing the app on the device failed
//////

エラーやワーニングこそ出ていないのですが、distにAPKが出力されません。
AIRSDKのバージョンは50.2.3で、元のが見つけられなかったFLEX_SDKはAIRSDK_Flex_Windowsの最新版をダウンロードして以下のように設定したのですが、これで大丈夫だったでしょうか？

set FLEX_SDK=E:\AIRSDK\AIRSDK_Flex_Windows

**Comment by ajwfrost** [comment]
Hi

The "FLEX_SDK" thing is a bit badly named I'm afraid, FlashDevelop used this variable name but they actually just meant the AIR SDK. So it looks like what you've done in terms of paths is correct, but if you are using just AIR (rather than Flex with AIR) then you need the normal AIRSDK_Windows package, rather than AIRSDK_Flex_Windows which is missing the compiler..

But then, the packaging part seems like the area that's going wrong. Are you able to install the AIR SDK Manager, and open this to the "Troubleshooting" page? This should then display the log when you try packaging up the application again, and we can then hopefully see the problem.

thanks

-------

「FLEX_SDK」という名前は少し悪い名前だと思いますが、FlashDevelop はこの変数名を使用しましたが、実際には AIR SDK を意味しているだけです。 したがって、パスに関して行ったことは正しいように見えますが、（AIR を含む Flex ではなく）AIR のみを使用している場合は、コンパイラが欠落している AIRSDK_Flex_Windows ではなく、通常の AIRSDK_Windows パッケージが必要です。

しかし、パッケージング部分が問題を抱えている領域のように思えます。 AIR SDK Manager をインストールして、「トラブルシューティング」ページを開いていただけますか? これにより、アプリケーションを再度パッケージ化しようとしたときにログが表示され、問題を確認できることが期待されます。

**Comment by HirokatsuIshikawa** [comment, marked as answer]
Thank you very much for all the explanations.

I immediately tried to find the problem using AIR SDK Manager, but even when I packaged it while displaying "Troubleshooting", nothing was displayed in "Troubleshooting".

Also, in the PC version, the application contents were executed by running the editor in an almost similar environment, but the package could not be created either and nothing was displayed in "Troubleshooting".

What are the conditions for using "Troubleshooting" in AIR SDK Manager?

Also, the OS is Windows 11.

----
各種説明どうもありがとうございます。

早速AIR SDK Managerを使用して問題個所を見つけようとしたのですが、"Troubleshooting"を表示しながらパッケージを行っても、"Troubleshooting"に何も表示されませんでした。

また、PC版の方はほぼ同様の環境でエディタの実行でアプリ内容が実行されましたが、こちらもパッケージが出来ず同様に"Troubleshooting"で何も表示されませんでした。

AIR SDK Managerで"Troubleshooting"を使用するための条件はどのようになっておりますでしょうか？

それと、OSはWindows11を使用しております。

**Comment by HirokatsuIshikawa** [comment]
Sorry, one of the problems seems to be that Java 20 deleted only the folder and remained in the app list without being treated as uninstalled. Troubleshooting will be done after officially uninstalling this from the app list. is now displayed.
Also, the output of the PC version now works fine.
It seems like the Android version is still a little lame, but I'll think about it a little more and move forward.
thank you very much.
----
すいません、こちらの不具合の一つがJava20がフォルダだけ消して、アンインストール扱いになっておらずアプリ一覧に残ったままだったのが原因のようで、これをアプリ一覧から正式にアンインストールしたらトラブルシューティングが表示されるようになりました。
また、PC版は出力うまくいくようになりました。
Android版はまだちょっとダメなようですが、もう少しこちらで考えて進めさせていただきます。
どうもありがとうございました。
